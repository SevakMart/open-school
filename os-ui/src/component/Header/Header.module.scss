@import "../../styles/Mixins/ButtonMixin.scss";
@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";

.mainContent {
  @include flexStyle(row, center, space-between);
  padding-left: 10.7%;
  padding-right: 9.5%;
  padding-top: 7%;
  background-color: #eeeff4;
  .leftContent {
    border: 2px solid blue;
    width: 40%;
    @include flexStyle(column, flex-start, space-between);
    h1 {
      display: inline;
      @include fontStyle("Montserrat", normal, 800, 48px, 59px);
      color: #5e617b;
      &::before {
        content: "Free ";
        color: #222427;
      }
    }
    p {
      display: inline;
      @include fontStyle("Montserrat", normal, 400, 16px, 24px);
      color: #5e617b;
    }
    .buttonContainer {
      width: 60%;
      @include flexStyle(row, center, space-between);
      & button:nth-of-type(1) {
        @include buttonStyle(
          #5e617b,
          #ffffff,
          16px,
          none,
          64px,
          32px,
          32px,
          22px,
          22px
        );
        @include fontStyle("Montserrat", normal, 500, 16px, 20px);
        text-transform: uppercase;
      }
      & button:nth-of-type(2) {
        @include buttonStyle(
          transparent,
          #5e617b,
          16px,
          none,
          64px,
          32px,
          32px,
          22px,
          22px
        );
        @include fontStyle("Montserrat", normal, 500, 16px, 20px);
        text-transform: uppercase;
        border-color: #5e617b;
      }
    }
  }
  .rightContent {
    border: 2px solid red;
    width: 50%;
    .arrowImage {
      & img:nth-of-type(2) {
        transform: translate(-75px, 10%);
      }
    }
  }
}
