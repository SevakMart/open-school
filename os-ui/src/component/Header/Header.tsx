import Navbar from '../Navbar/Navbar';
import Polygon from '../../svg/Polygon.svg';
import Rectangle from '../../svg/Rectangle.svg';
import styles from './Header.module.scss';
import {
  FREE_EDUCATIONAL_PLATFORM, SIGN_UP, HEADER_INTRODUCTION, SIGN_IN,
} from '../../constants/Strings';
import Button from '../Button/Button';

const Header = () => {
  const {
    mainContent, leftContent, rightContent, buttonContainer, arrowImage,
  } = styles;
  return (
    <>
      <Navbar />
      <div className={mainContent}>
        <div className={leftContent}>
          <h1>{FREE_EDUCATIONAL_PLATFORM}</h1>
          <p>{HEADER_INTRODUCTION}</p>
          <div className={buttonContainer}>
            <Button>{SIGN_UP}</Button>
            <Button>{SIGN_IN}</Button>
          </div>
        </div>
        <div className={rightContent}>
          <div className={arrowImage}>
            <img src={Polygon} alt="arrow icon" />
            <img src={Rectangle} alt="arrow icon" />
          </div>

        </div>
      </div>
    </>
  );
};
export default Header;
