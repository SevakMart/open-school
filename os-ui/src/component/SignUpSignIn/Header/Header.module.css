.closeIcon {
  position: absolute;
  top: 24px;
  right: 24px;
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}
.closeIcon:hover {
  background-color: #4BCEB5;
}

.headerContent {
  margin: 80px 80px 32px 80px;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
}
.headerContent h2 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 800;
  font-size: 2.25rem;
  line-height: 2.75rem;
  color: #222427;
}
.headerContent .iconContent {
  height: 36px;
  width: 35%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;
}
.headerContent .iconContent .icon {
  display: inline-block;
  width: 36px;
  height: 36px;
  cursor: pointer;
}

p {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 2.5rem;
  color: #5E617B;
}/*# sourceMappingURL=Header.module.css.map */