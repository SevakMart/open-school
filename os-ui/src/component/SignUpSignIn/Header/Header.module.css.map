{"version":3,"sources":["Header.module.scss","../../../styles/Mixins/FlexMixins.scss","Header.module.css","../../../styles/Configs/Colors.scss","../../../styles/Mixins/FontMixin.scss","../../../styles/Configs/Fonts.scss"],"names":[],"mappings":"AAMA;EACI,kBAAA;EACA,SAAA;EACA,WAAA;EACA,sBAAA;EACA,WAAA;EACA,YAAA;ECXF,aAAA;EACA,mBDWqB;ECVrB,mBDUyB;ECTzB,uBDSgC;EAC9B,eAAA;AEFJ;AFGI;EACI,yBGfU;ADclB;;AFIA;EACI,2BAAA;EACA,sBAAA;ECpBF,aAAA;EACA,sBDoBqB;ECnBrB,mBDmB4B;EClB5B,8BDkBmC;AEErC;AFDI;EItBF,yBCDY;EDEZ,kBJsBwC;EIrBxC,gBJqBgD;EIpBhD,kBJoBqD;EInBrD,oBJmB8D;EACxD,cGtBY;AD6BpB;AFLI;EACI,YAAA;EACA,UAAA;EC5BN,aAAA;EACA,mBD4ByB;EC3BzB,mBD2B6B;EC1B7B,6BD0BoC;AEUtC;AFTQ;EACI,qBAAA;EACA,WAAA;EACA,YAAA;EACA,eAAA;AEWZ;;AFPA;EItCE,yBCDY;EDEZ,kBJsCoC;EIrCpC,gBJqC4C;EIpC5C,kBJoCiD;EInCjD,mBJmC0D;EACxD,cGvCc;ADqDlB","file":"Header.module.css"}