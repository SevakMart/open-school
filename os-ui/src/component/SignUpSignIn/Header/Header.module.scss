@import "../../../styles/Configs/Colors.scss";
@import "../../../styles/Configs/Fonts.scss";
@import "../../../styles/Mixins/FlexMixins.scss";
@import "../../../styles/Mixins/ButtonMixin.scss";
@import "../../../styles/Mixins/FontMixin.scss";

.closeIcon {
    position: sticky;
    top:24px;
    margin-left: auto;
    right:24px;
    box-sizing: border-box;
    width: 24px;
    height: 24px;
    color: $email-icon-color;
    cursor: pointer;
    &:hover {
        color: brown;
    }
}
.headerContent {
    margin: 0px 50px 0px 50px;
    box-sizing: border-box;
    @include flexStyle(column,center,space-between);
    h2 {
        @include fontStyle($primary-font, normal, 800, 2.25rem, 2.75rem);
        color:$header-main-color;
    }
    .iconContent {
      height:36px;
      width:35%;
      @include flexStyle(row,center,space-evenly);
      .icon {
          display:inline-block;
          width:36px;
          height:36px;
          cursor: pointer;
      }
    }
}