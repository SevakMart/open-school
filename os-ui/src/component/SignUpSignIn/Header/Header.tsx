import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import { closeModal } from '../../../redux/Slices/PortalOpenStatus';
import LinkedinIcon1 from '../../../icons/LinkedinIcon/Linkedin1';
import EmailIcon1 from '../../../icons/EmailIcon/Email1';
import styles from './Header.module.scss';

const Header = ({
  mainTitle, shouldRemoveIconContent, isForgotPasswordContent, isVerificationContent,
}:{mainTitle:string, shouldRemoveIconContent:boolean, isForgotPasswordContent:boolean, isVerificationContent:boolean}) => {
  const {
    headerContent, iconContent, closeIcon, icon,
  } = styles;
  const dispatch = useDispatch();
  const { t } = useTranslation();

  const handleClosePortal = () => {
    dispatch(closeModal());
  };

  return (
    <>
      <div className={closeIcon} onClick={handleClosePortal}>âœ–</div>
      <div className={headerContent}>
        <h2>{mainTitle}</h2>
        {isVerificationContent && <p>{t('messages.verificationPageHint')}</p>}
        {isForgotPasswordContent
          && (
          <>
            <p>{t('string.forgotPsd.enterEmail')}</p>
            <p>{t('string.forgotPsd.sendNotification')}</p>
          </>
          )}
        { !shouldRemoveIconContent
        && (
        <>
          <div className={iconContent}>
            <div className={icon}><LinkedinIcon1 href="" /></div>
            <div className={icon}><EmailIcon1 title="example@example.com" /></div>
          </div>
          <p>{t('string.signIn.or')}</p>
        </>
        )}
      </div>
    </>
  );
};
export default Header;
