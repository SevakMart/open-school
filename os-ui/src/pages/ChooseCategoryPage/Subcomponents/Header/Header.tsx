import { useTranslation } from 'react-i18next';
import { Input } from '../../../../component/Input/Input';
import styles from './Header.module.scss';

const Header = ({ changeUrlQueries }:{changeUrlQueries:(title:string)=>void}) => {
  const { t } = useTranslation();
  const { mainContainer } = styles;
  return (
    <div className={mainContainer}>
      <h1>{t('string.homePage.header.chooseCategories')}</h1>
      <Input.SearchInput
        changeUrlQueries={(title:string) => changeUrlQueries(title)}
        className={['search', 'search__subcategoriesSearch']}
        placeholder={t('Search by name')}
      />
    </div>
  );
};
export default Header;
