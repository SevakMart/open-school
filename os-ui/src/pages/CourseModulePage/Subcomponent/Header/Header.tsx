import NavbarOnSignIn from '../../../../component/Navbar-Component/NavbarOnSignIn/NavbarOnSignIn';
import styles from './Header.module.scss';

const Header = () => {
  const { mainContainer } = styles;
  return (
    <div className={mainContainer}>
      <NavbarOnSignIn />
    </div>
  );
};
export default Header;
