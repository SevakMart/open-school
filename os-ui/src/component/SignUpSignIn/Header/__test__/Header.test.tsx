import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from '../../../../redux/Store';
import Header from '../Header';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create unit tests for Header component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Router>
        <Provider store={store}>
          <Header
            mainTitle="This is main Title"
            shouldRemoveIconContent={false}
            isForgotPasswordContent={false}
            isVerificationContent={false}
          />
        </Provider>
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
