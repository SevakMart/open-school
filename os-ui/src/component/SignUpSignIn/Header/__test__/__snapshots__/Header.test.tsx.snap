// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create unit tests for Header component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="closeIcon"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7,7 L17,17 M7,17 L17,7"
        fill="none"
        stroke="#000"
        stroke-width="2"
      />
    </svg>
  </div>
  <div
    class="headerContent"
  >
    <h2>
      This is main Title
    </h2>
    <div
      class="iconContent"
    >
      <div
        class="icon"
      >
        <a
          href=""
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(132, 138, 157); width: 100%; height: 100%;"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M847.7 112H176.3c-35.5 0-64.3 28.8-64.3 64.3v671.4c0 35.5 28.8 64.3 64.3 64.3h671.4c35.5 0 64.3-28.8 64.3-64.3V176.3c0-35.5-28.8-64.3-64.3-64.3zm0 736c-447.8-.1-671.7-.2-671.7-.3.1-447.8.2-671.7.3-671.7 447.8.1 671.7.2 671.7.3-.1 447.8-.2 671.7-.3 671.7zM230.6 411.9h118.7v381.8H230.6zm59.4-52.2c37.9 0 68.8-30.8 68.8-68.8a68.8 68.8 0 1 0-137.6 0c-.1 38 30.7 68.8 68.8 68.8zm252.3 245.1c0-49.8 9.5-98 71.2-98 60.8 0 61.7 56.9 61.7 101.2v185.7h118.6V584.3c0-102.8-22.2-181.9-142.3-181.9-57.7 0-96.4 31.7-112.3 61.7h-1.6v-52.2H423.7v381.8h118.6V604.8z"
            />
          </svg>
        </a>
      </div>
      <div
        class="icon"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(132, 138, 157); width: 100%; height: 100%;"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title />
          <path
            d="M24 5.457v13.909c0 .904-.732 1.636-1.636 1.636h-3.819V11.73L12 16.64l-6.545-4.91v9.273H1.636A1.636 1.636 0 0 1 0 19.366V5.457c0-2.023 2.309-3.178 3.927-1.964L5.455 4.64 12 9.548l6.545-4.91 1.528-1.145C21.69 2.28 24 3.434 24 5.457z"
          />
        </svg>
      </div>
    </div>
    <p>
      string.signIn.or
    </p>
  </div>
</DocumentFragment>
`;
